From 8fff4c264516bbf93af8f0379f64bffcda7b29ab Mon Sep 17 00:00:00 2001
From: Josh Wu <josh.wu@atmel.com>
Date: Mon, 17 Feb 2014 16:46:59 +0800
Subject: [PATCH] arm: at91: video: add new variable of output bpp for new LCD
 controller

so that we can support runtime change the output bpp.

Signed-off-by: Josh Wu <josh.wu@atmel.com>

Conflicts:
	board/atmel/sama5d4ek/sama5d4ek.c
---
 board/atmel/at91sam9n12ek/at91sam9n12ek.c | 1 +
 board/atmel/at91sam9x5ek/at91sam9x5ek.c   | 1 +
 board/atmel/sama5d3xek/sama5d3xek.c       | 1 +
 drivers/video/atmel_hlcdfb.c              | 7 +------
 include/lcd.h                             | 1 +
 5 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/board/atmel/at91sam9n12ek/at91sam9n12ek.c b/board/atmel/at91sam9n12ek/at91sam9n12ek.c
index 9adc992..2db0a80 100644
--- a/board/atmel/at91sam9n12ek/at91sam9n12ek.c
+++ b/board/atmel/at91sam9n12ek/at91sam9n12ek.c
@@ -85,6 +85,7 @@ vidinfo_t panel_info = {
 	.vl_row = 272,
 	.vl_clk = 9000000,
 	.vl_bpix = LCD_BPP,
+	.vl_bpox = LCD_OUTPUT_BPP,
 	.vl_sync = 0,
 	.vl_tft = 1,
 	.vl_hsync_len = 5,
diff --git a/board/atmel/at91sam9x5ek/at91sam9x5ek.c b/board/atmel/at91sam9x5ek/at91sam9x5ek.c
index 17a2a40..a1d44d8 100644
--- a/board/atmel/at91sam9x5ek/at91sam9x5ek.c
+++ b/board/atmel/at91sam9x5ek/at91sam9x5ek.c
@@ -116,6 +116,7 @@ vidinfo_t panel_info = {
 	.vl_clk = 24000000,
 	.vl_sync = LCDC_LCDCFG5_HSPOL | LCDC_LCDCFG5_VSPOL,
 	.vl_bpix = LCD_BPP,
+	.vl_bpox = LCD_OUTPUT_BPP,
 	.vl_tft = 1,
 	.vl_clk_pol = 1,
 	.vl_hsync_len = 128,
diff --git a/board/atmel/sama5d3xek/sama5d3xek.c b/board/atmel/sama5d3xek/sama5d3xek.c
index c835c12..67b7bb9 100644
--- a/board/atmel/sama5d3xek/sama5d3xek.c
+++ b/board/atmel/sama5d3xek/sama5d3xek.c
@@ -92,6 +92,7 @@ vidinfo_t panel_info = {
 	.vl_clk = 24000000,
 	.vl_sync = ATMEL_LCDC_INVLINE_NORMAL | ATMEL_LCDC_INVFRAME_NORMAL,
 	.vl_bpix = LCD_BPP,
+	.vl_bpox = LCD_OUTPUT_BPP,
 	.vl_tft = 1,
 	.vl_hsync_len = 128,
 	.vl_left_margin = 64,
diff --git a/drivers/video/atmel_hlcdfb.c b/drivers/video/atmel_hlcdfb.c
index 935ae42..79d22cd 100644
--- a/drivers/video/atmel_hlcdfb.c
+++ b/drivers/video/atmel_hlcdfb.c
@@ -96,11 +96,7 @@ void lcd_ctrl_init(void *lcdbase)
 
 	value |= panel_info.vl_sync;
 
-#ifndef LCD_OUTPUT_BPP
-	/* Output is 24bpp */
-	value |= LCDC_LCDCFG5_MODE_OUTPUT_24BPP;
-#else
-	switch (LCD_OUTPUT_BPP) {
+	switch (panel_info.vl_bpox) {
 	case 12:
 		value |= LCDC_LCDCFG5_MODE_OUTPUT_12BPP;
 		break;
@@ -117,7 +113,6 @@ void lcd_ctrl_init(void *lcdbase)
 		BUG();
 		break;
 	}
-#endif
 
 	value |= LCDC_LCDCFG5_GUARDTIME(ATMEL_LCDC_GUARD_TIME);
 	value |= (LCDC_LCDCFG5_DISPDLY | LCDC_LCDCFG5_VSPDLYS);
diff --git a/include/lcd.h b/include/lcd.h
index cc2ee3f..a4a5b34 100644
--- a/include/lcd.h
+++ b/include/lcd.h
@@ -155,6 +155,7 @@ typedef struct vidinfo {
 	u_long vl_tft;		/* 0 = passive, 1 = TFT */
 	u_long vl_cont_pol_low;	/* contrast polarity is low */
 	u_long vl_clk_pol;	/* clock polarity */
+	u_long vl_bpox;		/* Bits per pixel for output, can be 12, 16, 18, 24 */
 
 	/* Horizontal control register. */
 	u_long vl_hsync_len;	/* Length of horizontal sync */
-- 
1.9.1

